src/magicbox/ResultPreviewsManager.ts

```diff
@@ -78,6 +84,14 @@ export class ResultPreviewsManager {
   }
 
   public async displaySearchResultPreviewsForSuggestion(suggestion: HTMLElement) {
+    const externalOptions = this.getExternalOptions();
+    const currentDelay = (this.lastDelay = Utils.resolveAfter(
+      externalOptions.delayBeforePopulate == undefined ? 200 : externalOptions.delayBeforePopulate
+    ));
+    await currentDelay;
+    if (currentDelay !== this.lastDelay) {
```

Comment:
This `if` statement seems suspicious :) Is the goal to prevent two overlapping `displaySearchResultPreviewsForSuggestion` calls?

If so, how are we handling the case where two different suggestions are triggered with the same delay?

Reply:
Those are instances of `Promise` made on the spot by `Utils.resolveAfter`, they'll never be the same. Any object or GUID could be compared here. 

This delay, combined with this comparison, prevent `ResultPreviewsManager` from triggering more queries for previews than one every `delayBeforePopulate` milliseconds.